gcloud/monitoring/metric.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Metric Descriptors for the Google Monitoring API."""
+
+# Features intentionally omitted from this first version of the client library:
+#   - Creating and deleting metric descriptors.
+
+import collections
+
+from gcloud.monitoring.label import LabelDescriptor
+
+
+class MetricDescriptor(collections.namedtuple(
+        'MetricDescriptor', ('name type labels metric_kind value_type unit'
+                             ' description display_name'))):
+    """Specification of a metric type and its schema.
+
+    Metric descriptor instances are immutable.
+
+    :type name: string
+    :param name: The "resource name" of the metric descriptor. For example:
+                 ``"projects/<project_id>/metricDescriptors/<type>"``
+
+    :type type: string
+    :param type: The metric type including a DNS name prefix. For example:
+                 ``"compute.googleapis.com/instance/cpu/utilization"``
+
+    :type labels: list of :class:`~gcloud.monitoring.label.LabelDescriptor`
+    :param labels: A sequence of label descriptors specifying the labels used
+                   to identify a specific instance of this metric.
+
+    :type metric_kind: string
+    :param metric_kind: The kind of measurement. It must be one of ``"GAUGE"``,
```

Comment:
So, we would continue to work with strings internally but offer the user a set of named constants to use if they wish. I'll try it.